fault-tolerance/src/main/java/io/helidon/faulttolerance/AsyncImpl.java

```diff
@@ -35,14 +36,17 @@
         CompletableFuture<T> future = new CompletableFuture<>();
         AsyncTask<T> task = new AsyncTask<>(supplier, future);
 
+        Future<?> taskFuture;
         try {
-            executor.get().submit(task);
+            taskFuture = executor.get().submit(task);
         } catch (Throwable e) {
             // rejected execution and other executor related issues
             return Single.error(e);
         }
 
-        return Single.create(future);
+        Single<T> single = Single.create(future, true);
+        single.onCancel(() -> taskFuture.cancel(false));     // cancel task
```

Comment:
This doesn't look right. I think `Single` operators are still immutable and create a new instance, thus returing `single` will not have the cancel operation. You'd probably want `return single.onCancel(() -> taskFuture.cancel(true));`.

Reply:
I discussed this with @danielkec. Code works because of how we interact with `Future` but returning the `Single` from `onCancel` is the correct way. Fixed.